roles/keycloak/tasks/main.yml

```diff
@@ -0,0 +1,174 @@
+    release_namespace: "{{ keycloak_helm_release_namespace }}"
+    create_namespace: true
+    kubeconfig: /etc/kubernetes/admin.conf
+    wait: true
+    values: "{{ _keycloak_helm_values | combine(keycloak_helm_values, recursive=True) }}"
+
+- name: Disable pxc strict mode
+  community.mysql.mysql_query:
+    login_host: "{{ _pxc_service.resources[0].spec.clusterIP }}"
+    login_user: root
+    login_password: "{{ openstack_helm_endpoints.oslo_db.auth.admin.password }}"
+    query: "set global pxc_strict_mode='ENFORCING'"
+
+- name: Create realm and clients
+  when: (_keycloak_realms + keycloak_realms) | length > 0
+  block:
+    - name: Wait until Keycloak ready
+      kubernetes.core.k8s_info:
+        api_version: apps/v1
+        kind: StatefulSet
+        name: keycloak
+        namespace: "{{ keycloak_helm_release_namespace }}"
+      register: _keycloak_statefulest_info
+      until: _keycloak_statefulest_info.resources[0].status.readyReplicas >= 1
+      retries: 60
+      delay: 10
+
+    - name: Create realms
+      # Note(okozachenko1203): Ignoring changes in this task until the upstream issue
+      #                        https://github.com/keycloak/keycloak/issues/20970 is resolved.
+      changed_when: false
+      community.general.keycloak_realm:
+        auth_client_id: admin-cli
+        auth_keycloak_url: "https://{{ keycloak_host }}"
+        auth_realm: master
+        auth_username: "{{ keycloak_admin_username }}"
+        auth_password: "{{ keycloak_admin_password }}"
+        id: "{{ item.id }}"
+        realm: "{{ item.id }}"
+        enabled: true
+      loop: "{{ _keycloak_realms + keycloak_realms }}"
+      retries: 60
+      delay: 5
+      register: _result
+      until: _result is not failed
+
+    - name: Create clients
+      community.general.keycloak_client:
+        auth_client_id: admin-cli
+        auth_keycloak_url: "https://{{ keycloak_host }}"
+        auth_realm: master
+        auth_username: "{{ keycloak_admin_username }}"
+        auth_password: "{{ keycloak_admin_password }}"
+        state: present
+        realm: "{{ item.0.id }}"
+        client_id: "{{ item.1.id }}"
+        name: "{{ item.1.id }}"
+        root_url: "{{ item.1.root_url }}"
+        admin_url: "{{ item.1.admin_url }}"
+        base_url: "{{ item.1.base_url }}"
+        enabled: true
+        client_authenticator_type: client-secret
+        public_client: false
+        secret: "{{ item.1.secret }}"
+        redirect_uris: "{{ item.1.redirect_uris }}"
+        web_origins: "{{ item.1.web_origins }}"
+        standard_flow_enabled: "{{ item.1.standard_flow_enabled }}"
+        direct_access_grants_enabled: "{{ item.1.direct_access_grants_enabled }}"
+        protocol: "{{ item.1.protocol }}"
+      with_subelements:
+        - "{{ _keycloak_realms + keycloak_realms }}"
+        - clients
+      retries: 60
+      delay: 5
+      register: _result
+      until: _result is not failed
```

Comment:
I think it would be neat to move this to a role called `keycloak_client` similar to how we have `glance_image` .. in there, we would have defaults:

```yaml
keycloak_client_admin_username: "{{ keycloak_admin_username | default("admin") }}"
keycloak_client_admin_password: "{{ keycloak_admin_password }}"
```

by doing this, then we can just call that role whenever we need and dont need to have to put credentials into each service.